src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -570,32 +602,126 @@ export class ResultsViewPageStore {
-                };
-                return client.fetchAllClinicalDataInStudyUsingPOST({
-                    studyId:study.studyId,
-                    clinicalDataSingleStudyFilter: filter,
-                    clinicalDataType: "SAMPLE"
-                });
-            } else {
-                const filter: ClinicalDataMultiStudyFilter = {
-                    attributeIds: clinicalDataTypes,
-                    identifiers: this.samples.result.map((s: Sample) => ({entityId: s.sampleId, studyId: s.studyId}))
-                };
-                return client.fetchClinicalDataUsingPOST({
-                    clinicalDataType: "SAMPLE",
-                    clinicalDataMultiStudyFilter: filter
-                });
-            }
+        invoke: () => this.getClinicalData("SAMPLE", this.samples.result, ["CANCER_TYPE", "CANCER_TYPE_DETAILED"])
+    }, []);
+
+    private getClinicalData(clinicalDataType: "SAMPLE" | "PATIENT", entities: any[], attributeIds: string[]):
+    Promise<Array<ClinicalData>> {
+
+        // single study query endpoint is optimal so we should use it
+        // when there's only one study
+        if (this.studies.result.length === 1) {
+            const study = this.studies.result[0];
+            const filter: ClinicalDataSingleStudyFilter = {
+                attributeIds: attributeIds,
+                ids: _.map(entities, clinicalDataType === "SAMPLE" ? 'sampleId' : 'patientId')
+            };
+            return client.fetchAllClinicalDataInStudyUsingPOST({
+                studyId:study.studyId,
+                clinicalDataSingleStudyFilter: filter,
+                clinicalDataType: clinicalDataType
+            });
+        } else {
+            const filter: ClinicalDataMultiStudyFilter = {
+                attributeIds: attributeIds,
+                identifiers: entities.map((s: any) => clinicalDataType === "SAMPLE" ?
+                    ({entityId: s.sampleId, studyId: s.studyId}) : ({entityId: s.patientId, studyId: s.studyId}))
+            };
+            return client.fetchClinicalDataUsingPOST({
+                clinicalDataType: clinicalDataType,
+                clinicalDataMultiStudyFilter: filter
+            });
+        }
+    }
+
+    readonly survivalClinicalData = remoteData<ClinicalData[]>({
+        await: () => [
+            this.studies,
+            this.patients
+        ],
+        invoke: () => this.getClinicalData("PATIENT", this.patients.result, ["OS_STATUS", "OS_MONTHS", "DFS_STATUS", "DFS_MONTHS"])
+    }, []);
+
+    readonly survivalClinicalDataGroupByUniquePatientKey = remoteData<Dictionary<ClinicalData[]>>({
+        await: () => [
+            this.survivalClinicalData,
+        ],
+        invoke: async() => {
+            return _.groupBy(this.survivalClinicalData.result, 'uniquePatientKey');
+        }
+    });
+
+    private getPatientSurvivals(targetUniquePatientKeys: string[], statusAttributeId: string, monthsAttributeId: string,
```

Comment:
just for discussion: this method relies on async data but has no mechanism to wait for it.  this sort of bothers me.  one solution would be to just pass in the data (e.g. survivalClinicalDataGroupByUniquePatientKey and patient) so that the moxPromise would then be responisble for awaiting that data and feeding in to this method (which would become more pure)